### PR TITLE
Update Realtime Profiling Instance Naming

### PR DESCRIPTION
**Pull request by deleonenriqueta**
Resolves Issue [#1864](https://github.com/newrelic/newrelic-java-agent/issues/1864)

### CODE REVIEW

File: jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java [hunk 1]
```diff
@@ -29,4 +33,83 @@ void buildCommonAttributes() {
...
+    Attributes attributes = SetupUtils.buildCommonAttributes(config);
+
+    assertNotNull(attributes);
+    assertEquals("JFR", attributes.asMap().get(INSTRUMENTATION_NAME));
+    assertEquals("JFR-Uploader", attributes.asMap().get(INSTRUMENTATION_PROVIDER));
+    assertEquals("JFR-Uploader", attributes.asMap().get(COLLECTOR_NAME));
+    assertEquals("test_app_name", attributes.asMap().get(SERVICE_NAME));
+    assertEquals("test_app_name", attributes.asMap().get(APP_NAME));
+    assertEquals("test_hostname", attributes.asMap().get(HOSTNAME));
+    assertEquals("test_app_name", attributes.asMap().get(APP_NAME));
```
Comment (jtduffy) [inline jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java:52]: Looks to be a dup of the asserts above (APP_NAME and SERVICE_NAME checks)

File: jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java [hunk 2]
```diff
@@ -29,4 +33,83 @@ void buildCommonAttributes() {
...
+      Attributes attributes = SetupUtils.buildCommonAttributes(config);
+
+      assertNotNull(attributes);
+      assertEquals("JFR", attributes.asMap().get(INSTRUMENTATION_NAME));
+      assertEquals("JFR-Uploader", attributes.asMap().get(INSTRUMENTATION_PROVIDER));
+      assertEquals("JFR-Uploader", attributes.asMap().get(COLLECTOR_NAME));
+      assertEquals("test_app_name", attributes.asMap().get(SERVICE_NAME));
+      assertEquals("test_app_name", attributes.asMap().get(APP_NAME));
+      assertEquals("resolved_hostname", attributes.asMap().get(HOSTNAME));
+      assertEquals("test_app_name", attributes.asMap().get(APP_NAME));
```
Comment (jtduffy) [inline jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java:78]: Dups

File: jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java [hunk 3]
```diff
@@ -29,4 +33,83 @@ void buildCommonAttributes() {
...
+      Attributes attributes = SetupUtils.buildCommonAttributes(config);
+
+      assertNotNull(attributes);
+      assertEquals("JFR", attributes.asMap().get(INSTRUMENTATION_NAME));
+      assertEquals("JFR-Uploader", attributes.asMap().get(INSTRUMENTATION_PROVIDER));
+      assertEquals("JFR-Uploader", attributes.asMap().get(COLLECTOR_NAME));
+      assertEquals("test_app_name", attributes.asMap().get(SERVICE_NAME));
+      assertEquals("test_app_name", attributes.asMap().get(APP_NAME));
+      assertEquals(loopbackAddress.toString(), attributes.asMap().get(HOSTNAME));
+      assertEquals("test_app_name", attributes.asMap().get(APP_NAME));
```
Comment (jtduffy) [inline jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java:111]: Same duplication

File: jfr-daemon/src/main/java/com/newrelic/jfr/daemon/SetupUtils.java [hunk 4]
```diff
@@ -50,10 +50,14 @@ public static Attributes buildCommonAttributes(DaemonConfig config) {
             .put(AttributeNames.INSTRUMENTATION_PROVIDER, "JFR-Uploader")
             .put(AttributeNames.COLLECTOR_NAME, "JFR-Uploader");
     String hostname;
-    try {
-      hostname = InetAddress.getLocalHost().toString();
-    } catch (Throwable e) {
-      hostname = InetAddress.getLoopbackAddress().toString();
+    if (config.getHostname() == null) {
```
Comment (jtduffy) [inline jfr-daemon/src/main/java/com/newrelic/jfr/daemon/SetupUtils.java:53]: Not sure about having hostname as a config. Probably be good to get other opinions on it.

**Review by jtduffy — APPROVED** [review]
Resolve the duplicate test asserts and see what others think about the hostname config. 👍

File: jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java (hunk 1 above)
Comment (deleonenriqueta) [reply jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java:52]: updated

File: jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java (hunk 2 above)
Comment (deleonenriqueta) [reply jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java:78]: updated

File: jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java (hunk 3 above)
Comment (deleonenriqueta) [reply jfr-daemon/src/test/java/com/newrelic/jfr/daemon/SetupUtilsTest.java:111]: updated

File: jfr-daemon/src/main/java/com/newrelic/jfr/daemon/SetupUtils.java (hunk 4 above)
Comment (deleonenriqueta) [reply jfr-daemon/src/main/java/com/newrelic/jfr/daemon/SetupUtils.java:53]: Looking back, there is no real need to update the config. This value can be accessed and added to the commonAttributes from the Agent, without changing the config at all.

**Pull request closed without merging**